api4/config.go

```diff
@@ -156,6 +156,11 @@ func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 		*cfg.PluginSettings.MarketplaceURL = *appCfg.PluginSettings.MarketplaceURL
 	}
 
+	if cfg.PluginSettings.PluginStates[model.PluginIdFocalboard].Enable && cfg.FeatureFlags.BoardsProduct {
```

Comment:
Won't this prevent the server from starting if both versions are enabled at once? I'd be worried about preventing startup when we could just fail nicely by not running the plugin like we do below.

Reply:
This is just the updateConfig request handler. I tested with both enabled, and what would happen is in the `syncPluginsActiveState` method, it would detect that focalboard is enabled, and would disable and shut it down. Server startup goes on as usual. Can you clarify more on your concern?

Reply:
Oh, I see. I thought this was like the `Config.IsValid` method that prevents the server from starting if it returns an error